correccted resource location in azuredeploy.parameters.json

mddazure opened the pull request:
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,24 @@
+          "description": "Password for the Virtual Machine."
+        }
+      },
+      "windowsOSVersion": {
```
remove this and use the defaultValue in the template

bmoore-msft replied:
this is still needed

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "azfw",
```
remove defaultValue (increases surface area)

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "privateLinkServiceNetworkPolicies": "Enabled"
+            }
+        },
+        {
```
how is this used?

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat('https://', variables('storageAccounts_azfwdemodiag_name'), '.blob.core.windows.net/')]"
```
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat('https://', variables('storageAccounts_azfwdemodiag_name'), '.blob.core.windows.net/')]"
```
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat('https://', variables('storageAccounts_azfwdemodiag_name'), '.blob.core.windows.net/')]"
```
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_hub_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-88091e28-bfb2-40d7-a372-17daafc8cbc1.bastion.azure.com",
```
remove this property

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_on_prem_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-bc2e730d-498d-4f4d-828c-92888981428a.bastion.azure.com",
```
remove this property

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_spoke1_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-fd97b18a-2f11-4105-84f4-5dc4f4a912f1.bastion.azure.com",
```
remove this property

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -0,0 +1,1518 @@
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_spoke2_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-dd620c26-35a0-4473-ae4b-64c84600a8c8.bastion.azure.com",
```
remove this property

mddazure commented:
Incorporated changes requested, also fixed shared key for S2S VPN. Automtaed test for Public passes. Deployment test to Gov fails because come resource types used are not available in Gov yet.

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -84,7 +80,8 @@
         "virtualNetworkGateways_azfw_hub_gw_name": "azfw-hub-gw",
-        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw"    
+        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw",
+        "virtualNetworkGateways_shared_key":"[uniquestring(subscription().subscriptionId)]"    
```
what's the requirement for uniqueness here?  This will allow for only one per subscription...

mddazure replied:
Wanted to create a unique key for S2S connection. Subscr id as a seed may not be the best idea, you're right. Changed it to use deployment().name as the seed.

bmoore-msft commented on 101-azurefirewall-hub-spoke-demo/azuredeploy.json:
```diff
@@ -4,16 +4,12 @@
           },
           "adminPassword": {
-            "type": "securestring",
-            "metadata": {
```
revert this change (add the metadata property back)

mddazure replied:
Added metadata description back in the template and removed from the parameters file.
